src/main/java/org/jenkinsci/plugins/mesos/MesosCloud.java

```diff
@@ -147,6 +170,20 @@ public MesosCloud(
     }
   }
 
+  /**
+   * Copy constructor.
+   * Allows to create copies of the original mesos cloud. Since it's a singleton
+   * by design, this method also allows specifying a new name.
+   * @param name Name of the cloud to be created
+   * @param source Source Mesos cloud implementation
+   * @since TODO
+   */
+  public MesosCloud(@Nonnull String name, @Nonnull MesosCloud source) {
+      this(name, source.nativeLibraryPath, source.master, source.description, source.frameworkName,
+           source.slavesUser, source.principal, source.secret, source.slaveInfos, source.checkpoint,
+           source.onDemandRegistration, source.jenkinsURL);
+  }
```

Comment:
It's not a clone() operation, because we change the name.
Clone() call also presumes that we will clone all fields beneath, but we don't do it here.